src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.skia.cs

```diff
@@ -163,32 +153,190 @@ SkiaGeometrySource2D BuildGeometry()
 					spriteShape.FillBrush = null;
 				}
 
-				return geometry;
+				spriteShape.Geometry = compositor.CreatePathGeometry(path);
+				var shapeVisual = compositor.CreateShapeVisual();
+				shapeVisual.Shapes.Add(spriteShape);
+
+				sublayers.Add(shapeVisual);
+				parent.Children.InsertAtBottom(shapeVisual);
+
+				owner.ClippingIsSetByCornerRadius = cornerRadius != CornerRadius.None;
+				if (owner.ClippingIsSetByCornerRadius)
+				{
+					parent.Clip = new CompositionGeometricClip() { Geometry = spriteShape.Geometry };
+				}
 			}
+			else
+			{
+				var shapeVisual = compositor.CreateShapeVisual();
 
-			spriteShape.Geometry = compositor.CreatePathGeometry(new CompositionPath(BuildGeometry()));
+				var backgroundShape = compositor.CreateSpriteShape();
 
-			shapeVisual.Size = new Vector2((float)area.Width, (float)area.Height);
-			shapeVisual.Offset = new Vector3(0, 0, 0);
+				if (background is GradientBrush gradientBackground)
+				{
+					var fullArea = new Rect(
+						area.X + borderThickness.Left,
+						area.Y + borderThickness.Top,
+						area.Width - borderThickness.Left - borderThickness.Right,
+						area.Height - borderThickness.Top - borderThickness.Bottom);
 
-			shapeVisual.Shapes.Add(spriteShape);
+					var insideArea = new Rect(default, fullArea.Size);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'insideArea'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5924508)